src/components/app/data/hooks/useIsBFFEnabled.test.jsx

```diff
@@ -0,0 +1,53 @@
+import { renderHook } from '@testing-library/react-hooks';
+import { QueryClientProvider } from '@tanstack/react-query';
+import { MemoryRouter, useLocation } from 'react-router-dom';
+import { AppContext } from '@edx/frontend-platform/react';
+import useIsBFFEnabled from './useIsBFFEnabled';
+import { queryEnterpriseLearnerDashboardBFF, resolveBFFQuery } from '../queries';
+import { queryClient } from '../../../../utils/tests';
+import { authenticatedUserFactory } from '../services/data/__factories__';
+
+jest.mock('../queries', () => ({
+  ...jest.requireActual('../queries'),
+  resolveBFFQuery: jest.fn(),
+}));
+jest.mock('react-router-dom', () => ({
+  ...jest.requireActual('react-router-dom'),
+  useLocation: jest.fn(),
+}));
+const mockAuthenticatedUser = authenticatedUserFactory();
+
+describe('useIsBFFEnabled', () => {
+  const Wrapper = ({ routes = null, children }) => (
+    <QueryClientProvider client={queryClient()}>
+      <MemoryRouter initialEntries={[routes]}>
+        <AppContext.Provider value={{ authenticatedUser: mockAuthenticatedUser }}>
+          {children}
+        </AppContext.Provider>
+      </MemoryRouter>
+    </QueryClientProvider>
+  );
+  beforeEach(() => {
+    jest.clearAllMocks();
+    useLocation.mockReturnValue({ pathname: '/test-enterprise' });
```

Comment:
nit: shouldn't be a need to mock `useLocation` if rendering a real `MemoryRoute`.